src/rules/css/invalidNesting.js

```diff
@@ -1,10 +1,24 @@
 import * as messages from 'messages';
 
+import { basicCompatVersionComparison } from '../../utils';
+
+// Allowed version is 117.0 and above
+const CSS_NESTING_MIN_VERSION = 116;
```

Comment:
I don't think this is correct, though I understand why you set this constant to `117 - 1`. I think this would work a bit better, though:

```diff
diff --git a/src/rules/css/invalidNesting.js b/src/rules/css/invalidNesting.js
index b8cf9733..b51eab8f 100644
--- a/src/rules/css/invalidNesting.js
+++ b/src/rules/css/invalidNesting.js
@@ -3,7 +3,7 @@ import * as messages from 'messages';
 import { basicCompatVersionComparison } from '../../utils';
 
 // Allowed version is 117.0 and above
-const CSS_NESTING_MIN_VERSION = 116;
+const CSS_NESTING_MIN_VERSION = 117;
 
 export function invalidNesting(
   cssNode,
@@ -11,9 +11,10 @@ export function invalidNesting(
   { startLine, startColumn, addonMetadata } = {}
 ) {
   if (
-    basicCompatVersionComparison(
-      addonMetadata?.firefoxMinVersion,
-      CSS_NESTING_MIN_VERSION
+    addonMetadata?.firefoxMinVersion &&
+    !basicCompatVersionComparison(
+      CSS_NESTING_MIN_VERSION,
+      addonMetadata?.firefoxMinVersion
     )
   ) {
     return [];
diff --git a/tests/unit/rules/css/test.invalidNesting.js b/tests/unit/rules/css/test.invalidNesting.js
index ecf05ea4..4ec67b11 100644
--- a/tests/unit/rules/css/test.invalidNesting.js
+++ b/tests/unit/rules/css/test.invalidNesting.js
@@ -5,28 +5,31 @@ import { VALIDATION_WARNING } from 'const';
 import CSSScanner from 'scanners/css';
 
 describe('CSS Rule InvalidNesting', () => {
-  it('should detect invalid nesting', async () => {
-    const code = oneLine`/* I'm a comment */
+  it.each(['60.5', '116.0', '116'])(
+    'should detect invalid nesting when firefoxMinVersion=%s',
+    async (firefoxMinVersion) => {
+      const code = oneLine`/* I'm a comment */
       #something {
         .bar {
           height: 100px;
         }
       }`;
-    const cssScanner = new CSSScanner(code, 'fakeFile.css', {
-      addonMetadata: {
-        firefoxStrictMinVersion: '60.5',
-      },
-    });
+      const cssScanner = new CSSScanner(code, 'fakeFile.css', {
+        addonMetadata: {
+          firefoxMinVersion,
+        },
+      });
 
-    const { linterMessages } = await cssScanner.scan();
-    expect(linterMessages.length).toEqual(1);
-    expect(linterMessages[0].code).toEqual(
-      messages.INVALID_SELECTOR_NESTING.code
-    );
-    expect(linterMessages[0].type).toEqual(VALIDATION_WARNING);
-  });
+      const { linterMessages } = await cssScanner.scan();
+      expect(linterMessages.length).toEqual(1);
+      expect(linterMessages[0].code).toEqual(
+        messages.INVALID_SELECTOR_NESTING.code
+      );
+      expect(linterMessages[0].type).toEqual(VALIDATION_WARNING);
+    }
+  );
 
-  it.each(['117.0.0', '117.0.1'])(
+  it.each(['117', '117.0', '117.1', '117.0.0', '117.0.1'])(
     'should not report invalid nesting when firefoxMinVersion=%s',
     async (firefoxMinVersion) => {
       const code = oneLine`/* I'm a comment */
```

Reply:
I applied the changes and added the test cases, where the order of `basicCompatVersionComparison` is relevant and breaks.
The second argument cannot be a string with complex version formats, it will get casted to a `number`. This worked in the special cases of Major versions (`116`) and Major and Minor versions (`116.1`), but not for Major, Minor and Patch Versions.

How do we proceed?

Reply:
Ha. Yeah, now I get it... We should pass `firefoxMinVersion` as second parameter like it is done here: https://github.com/mozilla/addons-linter/blob/0d3e83b8e576dc26b92b369c2debf83697a790d6/src/parsers/manifestjson.js#L158

That said, I overlooked an important detail: `minFirefoxVersion` here is processed here: https://github.com/mozilla/addons-linter/blob/0d3e83b8e576dc26b92b369c2debf83697a790d6/src/parsers/manifestjson.js#682

That's why it doesn't work like it should... What do you think of the following diff?

```diff
diff --git a/src/parsers/manifestjson.js b/src/parsers/manifestjson.js
index b160062b..7b4dae8f 100644
--- a/src/parsers/manifestjson.js
+++ b/src/parsers/manifestjson.js
@@ -1293,6 +1293,7 @@ export default class ManifestJSONParser extends JSONParser {
    * @property {number} type
    * @property {string} version
    * @property {string} firefoxMinVersion
+   * @property {string} firefoxStrictMinVersion
    * @property {Set<string>} experimentApiPaths
    *
    * @returns {Metadata}
@@ -1304,10 +1305,14 @@ export default class ManifestJSONParser extends JSONParser {
       name: this.parsedJSON.name,
       type: PACKAGE_EXTENSION,
       version: this.parsedJSON.version,
+      // This is the `strict_min_version` value set in the `manifest.json` file
+      // for Firefox for desktop.
       firefoxMinVersion:
         this.parsedJSON.applications &&
         this.parsedJSON.applications.gecko &&
         this.parsedJSON.applications.gecko.strict_min_version,
+      // This is the strict min *major* version for Firefox for desktop.
+      firefoxStrictMinVersion: firefoxStrictMinVersion(this.parsedJSON),
       experimentApiPaths: this.getExperimentApiPaths(),
     };
   }
diff --git a/src/rules/css/invalidNesting.js b/src/rules/css/invalidNesting.js
index b8cf9733..cd6d22de 100644
--- a/src/rules/css/invalidNesting.js
+++ b/src/rules/css/invalidNesting.js
@@ -3,7 +3,7 @@ import * as messages from 'messages';
 import { basicCompatVersionComparison } from '../../utils';
 
 // Allowed version is 117.0 and above
-const CSS_NESTING_MIN_VERSION = 116;
+const CSS_NESTING_MIN_VERSION = 117;
 
 export function invalidNesting(
   cssNode,
@@ -11,9 +11,10 @@ export function invalidNesting(
   { startLine, startColumn, addonMetadata } = {}
 ) {
   if (
-    basicCompatVersionComparison(
-      addonMetadata?.firefoxMinVersion,
-      CSS_NESTING_MIN_VERSION
+    addonMetadata?.firefoxStrictMinVersion &&
+    !basicCompatVersionComparison(
+      CSS_NESTING_MIN_VERSION,
+      addonMetadata?.firefoxStrictMinVersion
     )
   ) {
     return [];
diff --git a/tests/unit/rules/css/test.invalidNesting.js b/tests/unit/rules/css/test.invalidNesting.js
index ecf05ea4..2ad372f4 100644
--- a/tests/unit/rules/css/test.invalidNesting.js
+++ b/tests/unit/rules/css/test.invalidNesting.js
@@ -5,30 +5,33 @@ import { VALIDATION_WARNING } from 'const';
 import CSSScanner from 'scanners/css';
 
 describe('CSS Rule InvalidNesting', () => {
-  it('should detect invalid nesting', async () => {
-    const code = oneLine`/* I'm a comment */
+  it.each([60, 116])(
+    'should detect invalid nesting when firefoxStrictMinVersion=%d',
+    async (firefoxStrictMinVersion) => {
+      const code = oneLine`/* I'm a comment */
       #something {
         .bar {
           height: 100px;
         }
       }`;
-    const cssScanner = new CSSScanner(code, 'fakeFile.css', {
-      addonMetadata: {
-        firefoxStrictMinVersion: '60.5',
-      },
-    });
+      const cssScanner = new CSSScanner(code, 'fakeFile.css', {
+        addonMetadata: {
+          firefoxStrictMinVersion,
+        },
+      });
 
-    const { linterMessages } = await cssScanner.scan();
-    expect(linterMessages.length).toEqual(1);
-    expect(linterMessages[0].code).toEqual(
-      messages.INVALID_SELECTOR_NESTING.code
-    );
-    expect(linterMessages[0].type).toEqual(VALIDATION_WARNING);
-  });
+      const { linterMessages } = await cssScanner.scan();
+      expect(linterMessages.length).toEqual(1);
+      expect(linterMessages[0].code).toEqual(
+        messages.INVALID_SELECTOR_NESTING.code
+      );
+      expect(linterMessages[0].type).toEqual(VALIDATION_WARNING);
+    }
+  );
 
-  it.each(['117.0.0', '117.0.1'])(
-    'should not report invalid nesting when firefoxMinVersion=%s',
-    async (firefoxMinVersion) => {
+  it.each([117, 118])(
+    'should not report invalid nesting when firefoxStrictMinVersion=%d',
+    async (firefoxStrictMinVersion) => {
       const code = oneLine`/* I'm a comment */
       #something {
         .bar {
@@ -37,7 +40,7 @@ describe('CSS Rule InvalidNesting', () => {
       }`;
       const cssScanner = new CSSScanner(code, 'fakeFile.css', {
         addonMetadata: {
-          firefoxMinVersion,
+          firefoxStrictMinVersion,
         },
       });
       const { linterMessages } = await cssScanner.scan();
```

Reply:
Oh, I didn't know, there already was a function to extract the major version! Sweet!